### ISSUE TITLE
Exception in ArmoredOutputStream caused by missing version info in NuGet package

### ISSUE THREAD

**Issue by christerk** [bcgit/bc-csharp#495]
The 2.2.1 release file that is downloaded by the Visual Studio NuGet manager is missing version information, which causes the constructor of the ArmouredOutputStream class to generate a null reference exception due to the GetVersion() method trying to access Assembly information.

It would be great to get a new dll pushed to NuGet to correct this. 1.9.0 does not have this issue.

**Comment by niemyjski** [comment]
I'm also running into this, issue

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Org.BouncyCastle.Bcpg.ArmoredOutputStream.CreateVersion()
   at Org.BouncyCastle.Bcpg.ArmoredOutputStream..cctor()
   ```
   
 ```
 private static string CreateVersion()
    {
      Assembly executingAssembly = Assembly.GetExecutingAssembly();
      return executingAssembly.GetCustomAttribute<AssemblyTitleAttribute>().Title + " v" + executingAssembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
    }
  ```
  
  I don't even know why this version information would be required.

**Comment by niemyjski** [comment]
@peterdettman I don't see how this is ever going to work. That's the current executing assembly from nuget.. `/bin/Debug/net6.0/BouncyCastle.Crypto.dll`

![image](https://github.com/bcgit/bc-csharp/assets/1020579/93f957af-9770-46ae-aa80-ee1b86d4bc26)

**Comment by peterdettman** [comment]
@niemyjski  What NuGet package are you referencing? Our official NuGet package is here: https://www.nuget.org/packages/BouncyCastle.Cryptography . I would guess maybe you are using BouncyCastle.NetCore?

In particular the DLL name in our NuGet is BouncyCastle.Cryptography.dll (not BouncyCastle.Crypto.dll), and the assembly contains 16 custom attributes (including the ones we rely on here). FYI, this information is used to add a header indicating which software generated a given file. In a test project (net6.0) using the official package, the CreateVersion function finds both attributes and returns "BouncyCastle.NET Cryptography (net6.0) v2.2.1+b9c0074fb1".

The assembly attributes are added at build time for us (Nerdbank.GitVersioning), so perhaps this is an oversight in a third party build. I am trying to release 2.3.0 at the moment; I have changed the code to at least not crash if the attributes aren't found and use a default value instead ("BouncyCastle (unknown version)").

**Comment by peterdettman** [comment]
Addressed in 2.3.0.
